doc/design/selected_rows.md

```diff
@@ -0,0 +1,56 @@
+# Design Doc: Selected Rows
+
+`SelectedRows` is a kind of sparse tensor data type, which is designed to support `embedding` operators. The gradient of embedding table is a sparse tensor. Only a few rows are non-zero values in that tensor. It is straightforward to represent the sparse tensor by the following sparse tensor data structure:
+
+```cpp
+class SelectedRows {
+ private:
+  vector<int> rows_;
+  Tensor value_;
+  int height_;
+};
+```
+
+The field `height_` shows the first dimension of `SelectedRows`. The `rows` are the indices of which rows of `SelectedRows` are non-zeros. The `value_` field is an N-dim tensor and shape is `[NUM_ROWS, ...]`, which supplies values for each row. The dimension of `SelectedRows` satisfies `[height_] + value_.shape[1:]`.
```

Comment:
shape is `[NUM_ROWS, ...]`  ==> shape is `[rows.size() /* NUM_ROWS */, ...]`

Reply:
```shell
height_  --> rows_number
rows  --> non_zero_row_idxs
value_  --> non_zero_rows
```
seems easier for read code and understanding.

Reply:
Done.